honeybee_radiance_folder/folder.py

```diff
@@ -730,6 +734,82 @@ def octree_scene_mapping(self):
+        """List of grids for each light path. The light paths are grouped as two phase,
+        three phase, and five phase. Two phase consist of static apertures and aperture
+        groups without a tmtx key in their states. The rest, aperture groups with a tmtx
+        key in their states, are added to three phase. Five phase is a copy of three
+        phase."""
+
+        two_phase, three_phase = [], []
+
+        grid_info = self.grid_info()
+
+        states = self.aperture_groups_states(full=True)
+        mtx_groups = []
+        non_mtx_groups = []
+        # get list of mtx groups and non-mtx groups
+        for aperture_group, ap_states in states.items():
+            for state in ap_states:
+                if aperture_group in mtx_groups or aperture_group in non_mtx_groups:
+                    continue
+                if 'tmtx' in state:
+                    mtx_groups.append(aperture_group)
+                else:
+                    non_mtx_groups.append(aperture_group)
+
+        two_phase_dict = dict()
+        three_phase_dict = dict()
+
+        for grid in grid_info:
+            light_paths = grid['light_path']
+            for light_path in light_paths:
+                light_path = light_path[0]
+                if light_path in non_mtx_groups:
+                    if light_path in two_phase_dict:
+                        two_phase_dict[light_path].append(grid)
+                    else:
+                        two_phase_dict[light_path] = [grid]
+                elif light_path in mtx_groups:
+                    if light_path in three_phase_dict:
+                        three_phase_dict[light_path].append(grid)
+                    else:
+                        three_phase_dict[light_path] = [grid]
+                else:
+                    # static apertures
+                    if '__static_apertures__' in two_phase_dict:
+                        two_phase_dict['__static_apertures__'].append(grid)
+                    else:
+                        two_phase_dict['__static_apertures__'] = [grid]
+        
+        for light_path, grids in two_phase_dict.items():
+                two_phase.append(
+                    {
+                        'identifier': light_path,
+                        'grid': grids
+                    }
+                )
+        for light_path, grids in three_phase_dict.items():
+                three_phase.append(
+                    {
+                        'identifier': light_path,
+                        'grid': grids
+                    }
+                )
+
+        grid_mapping = {
+            'two_phase': two_phase,
+            'three_phase': three_phase,
+            'five_phase': three_phase # same as three phase
+        }
+
+        grid_mapping_file = os.path.join(self.folder, 'grid_mapping.json')
+
+        with open(grid_mapping_file, 'w') as outf:
+            outf.write(json.dumps(grid_mapping, indent=2))
+
+        return grid_mapping
```

Comment:
You are aware of the context better but don't we need to write these as 3 separate files?

Reply:
I did this initially but I don't really use the files, since I use the output directly in `honeybee-radiance`, as in:
`grid_mapping = model_folder.grid_mapping()`

Reply:
But if you think 3 separate files are better I can do that.

Reply:
If it works for you and in the recipes, then it works for me!